index.js

```diff
@@ -1,33 +1,212 @@
+        schema_stream_retriever = schema_data;
+        if(wrapper_access_token === null){
+            setGlobalAccessTokenAndProcessResources(records, schema_data);
+        }else{
+          processResources(records, schema_data);
+        }
+      })
+    .catch(function(e){
+      console.log(e, e.stack);
+    });
+  }else{
+    if(wrapper_access_token === null){
+      setGlobalAccessTokenAndProcessResources(records, schema_stream_retriever);
+    }else{
+      processResources(records, schema_stream_retriever);
+    }
+  }
+}
+
+var setGlobalAccessTokenAndProcessResources = (function(records, schema_data){
+  set_wrapper_access_token()
+        .then(function(access_token){
+          console.log('Access token is - ' + access_token);
+          wrapper_access_token = access_token;
+          processResources(records, schema_data);
+        });
+  });
+
+//process resources
+var processResources = function(records, schema_data){
+  records.forEach(function(record){
+      var data = record.kinesis.data;
+      var json_data = avro_decoded_data(schema_data, data);
+      if(config.get('isABib')){
+        resource_in_detail(json_data.id, true)
+          .then(function(bib) {
+            var stream = config.get('bib.stream');
+            postResourcesToStream(bib, stream);
+          })
+          .catch(function (e){
+            console.log(e);
+          });
+      }else{
+        resource_in_detail(json_data.id, false)
+          .then(function(item){
+            var stream = config.get('item.stream');
+            postResourcesToStream(item, stream);
+          })
+      }
+    });
+}
 
+//get schema
+var request = require('request');
+var schema = function(url) {
+  return new Promise(function(resolve, reject) {
+      console.log('Querying for schema - ' + url);
+      request(url, function(error, response, body) {
+        if(!error && response.statusCode == 200){
+          resolve(JSON.parse(body).data.schema);
+        }else{
+          console.log("An error occurred - " + response.statusCode);
+          reject("Error occurred while getting schema - " + response.statusCode);
+        }
+      })
+    })
+}
 
+//use avro to deserialize
+var avro_decoded_data = function(schema_data, record){
+  const type = avro.parse(schema_data);
+  var decoded = new Buffer(record, 'base64');
+  var verify = type.fromBuffer(decoded);
+  return JSON.parse(verify);
+}
 
+//get full bib/item details for each bib/item id
+var resource_in_detail = function(id, isBib){
+  return new Promise(function(resolve, reject){
+    var operation = retry.operation({
+        retries: 5,
+        factor: 3,
+        minTimeout: 1 * 1000,
+        maxTimeout: 60 * 1000,
+        randomize: true,
+    });
+    operation.attempt(function(currentAttempt) {
+      if(isBib){
+        console.log('Requesting for bib info');
+        wrapper.requestSingleBib(id, (errorBibReq, results) => {
+        getResult(errorBibReq, results, true, id, operation, currentAttempt)
```

Comment:
@EdwinGuzman,
Thanks. I have made the changes and pushed. We can avoid these issues of consistency by using like a check style to ensure that everyone's code style (indentation/spacing) is uniform.